[ENH] BEP045: "Peripheral" physiology (raw data)

smoia opened the pull request:
This BEP proposes an expansion of the raw physiological data section of the standard.

The main development is happening in https://github.com/physiopy/bids-specification-physio

To see the rule of engagements, see [the discussion board](https://github.com/physiopy/bids-specification-physio/discussions/1).

To leave comments, engage with the physiopy repository (specifically, [with its PR #32](https://github.com/physiopy/bids-specification-physio/pull/32/)) rather than with this PR.
However, we're not really ready for reviews or comments yet, we are opening a draft just to have the HTML and schema json preview. We will communicate ASAP when we are ready for public engagement.

m-miedema committed:
Update datatypes.yaml

I'm not sure if a corresponding change should be made to modalities.yaml, I'm having trouble finding where the latter table is used by the compiler macros.

ShiyuBelleWang committed:
Incorporate physio BEP section 4

ShiyuBelleWang committed:
Update macros

ShiyuBelleWang committed:
Update physiological-recordings.md

ShiyuBelleWang committed:
Update physiological-recordings.md

ShiyuBelleWang committed:
Update physiological-recordings.md macros

m-miedema committed:
Migrate new/updated metadata fields relevant for raw data

Some checks on formatting (particularly for how I entered the allowable values) might be necessary here, I also added some comments on the BEP document to flag potential issues or changes we could make going forward.

m-miedema committed:
Migrate new/updated metadata fields relevant for raw data

Some checks on formatting (particularly for how I entered the allowable values) might be necessary here, I also added some comments on the BEP document to flag potential issues or changes we could make going forward.

m-miedema committed:
Addresses #21: adds PhysioTypeEnriched

m-miedema committed:
Addresses #21: adds PhysioTypeEnriched

m-miedema committed:
Added MeasureType entries to enums, updated metadata accordingly.

effigies reviewed (commented):
Should at least partially address the validation failures.

(I'm assuming that technical schema comments are more appropriate here than the regular discussion channels you mentioned. LMK if you'd rather they go somewhere else.)

ShiyuBelleWang committed:
Update physiological recordings MACROS_filetree

ShiyuBelleWang committed:
Remove repetitive examples

Removed example file structure for physiological recordings.

ShiyuBelleWang committed:
Add columns table for physiological recordings metadata

ShiyuBelleWang committed:
Add MeasureType descriptions section

Add MeasureType descriptions section to documentation

ShiyuBelleWang committed:
Update physiological recordings documentation

Removed unnecessary code blocks and clarified recommendations for recording physiological data.

ShiyuBelleWang committed:
Add MeasureType descriptions table

Added descriptions for various MeasureTypes related to physiological recordings.

scott-huberty commented on src/modality-specific-files/physiological-recordings.md:
```diff
@@ -22,162 +16,210 @@ JSON file for storing metadata fields (see below).
-For any other data to be specified in columns, the column names can be chosen
-as deemed appropriate by the researcher.
+- The most important one is `MeasureType`, a **RECOMMENDED** metadata that indicates the actual nature of the data in the column. 
+    - This metadata value is a string that **MUST** come from a set of keywords (see table 2.2).
```
```suggestion
    - This metadata value is a string that **MUST** come from a set of keywords (see table 2.3).
```

In the rendered [PDF](https://circleci-tasks-prod.s3.us-east-1.amazonaws.com/forks/storage/artifacts/100f87fd-e971-4c5c-9ee6-f1ab7b339342/968156422/e9c509a6-a674-41eb-b684-47d2529f82ed/0/bids-spec.pdf?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Checksum-Mode=ENABLED&X-Amz-Credential=ASIAQVFQINEOA2JVV7V4%2F20260219%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20260219T171808Z&X-Amz-Expires=60&X-Amz-Security-Token=IQoJb3JpZ2luX2VjELr%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLWVhc3QtMSJHMEUCIE28oojkKTgKZCybqfYhJIWr8hMnWkgFDM1KLpJo8ynrAiEArfcM8xe1E67H09GihgKsaqg5ytFslDIyeT7FL7ZDk64qtAIIgv%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARADGgwwNDU0NjY4MDY1NTYiDILDBad38g%2BrE0Pw7iqIAmNSghsk2fz0FmoFRNHeTNKfW%2BMbWfHB8APIWnJva5hH32%2BapInaCf8S2Qx3%2BqaQBNiCiGasw4mvy%2FEi7IeoBdE3mqAqFVp9s3RKUGPYtuvd1iq2rSqFPmNH2P07McRjektxw9B%2F9azgDI2yt5R37sQs85v7B96qSojDU68tozqrEolQOdU8UFSlrNA5SfO%2BoSFkRc5eBuZH4otZSINbALOARx2%2F4idCFcu8rX7cIv9orFGUx7pGEEuwmX5cs5fDUBUkRlAQZeX4ythjJhwfsXzdj3qSH9JYj4NqtvD%2FMelqaHZlNNl9iHCsW6eLa5FTV64pagfDudiIRrEs33eAoSHlX%2BoSdglGHzCvh93MBjqdAf10rSxv3pySApT1PMyaejWJ%2BOSqXlvliiO0dKuOiVWXYIDlKNy6Hsjck9KMUzPUk40nFkEqorDw2LP7ffnMnLDQVwAhxTGfyqnnxqLub1MT7intu02jACJsUn0WtMgXE4MetPhe0eIataJpve%2Ftr%2Bn9HS%2B5elT44Ak2Du1pU0NZP2FzPCm5i7PeUU5SRrpF77VoZsbfj%2FJFNiESclY%3D&X-Amz-SignedHeaders=host&x-id=GetObject&X-Amz-Signature=09e6b666a7035ad93413876c8aa34513de2b1e79c42619e836b9d0edae083f83) the Table for `MeasureType` appears to be table 2.3

oesteban commented:
> sorry not sure why it took so long to get on the same page there..

I think BIDS' specifications could be clearer, no need to say sorry ;)

> Is it just me that finds it very confusing that eye-tracking is under the physiology modality -> generic physiology TSV files do not constrain column names (and hence require the `Columns` key)

That's exactly what BEP020 contributed to BIDS. It provides a pathway to control for column names and metadata by providing a `PhysioType`, which is a controlled vocabulary currently having only `generic` and `eyetrack`. Adding to this vocabulary allows having other data types define their columns' control as well, without the need for spawning arbitrary `<datatype>/` folders (like `func/`).

smoia committed:
Update recommendations for physiological recordings

Clarified recommendations for file separation based on recording devices and metadata. Emphasized the importance of interpreting metadata correctly.

me-pic committed:
Change specified to enriched

ShiyuBelleWang committed:
Reformat & remove redundancy

Updated headings and removed redundant description of "generic" Physiotype

bendichter commented on src/modality-specific-files/physiological-recordings.md:
````diff
@@ -429,6 +505,111 @@ the `OnsetSource` is set to `"n/a"` in `sub-01_task-nback_physioevents.json`:
+}
+```
+
+As described in the table below ("Metadata fields for column description."), this BEP is adding a few metadata to describe columns.
````
I think it is against convention to mention the BEP here.
